client/src/lib.rs

```diff
@@ -426,7 +426,8 @@ pub fn handle_program_log<T: anchor_lang::Event + anchor_lang::AnchorDeserialize
 pub fn handle_system_log(this_program_str: &str, log: &str) -> (Option<String>, bool) {
     if log.starts_with(&format!("Program {this_program_str} log:")) {
         (Some(this_program_str.to_string()), false)
-    } else if log.contains("invoke") {
+    } else if log.contains("invoke") && !log.ends_with("[1]") {
+        // `Invoke [1]` instructions are pushed to the stack in `parse_logs_response`, so this ensures we only push CPIs to the stack at this stage
```

Comment:
Could you keep lines <100 chars? Commenting only here but applies to all other comments in this PR too.

Reply:
Changelog added and lines have been shortened